validation/lagom-validation/course-management.md

```diff
@@ -0,0 +1,54 @@
+# Course Validation
+Every invalid equivalence class returns an error. Every valid class returns an empty sequence.
+
+## Course
+
+### courseId (String)
+Not validated, as it is a generated UUID.
+
+### courseName (String)
+- vCourseName1: courseName.length ∈ [1, 100]
+- iCourseName1: courseName is the empty String
+- iCourseName2: courseName.length ∉ 100
+
+### courseType (String)
+*validCourseTypes* := {"Lecture", "Seminar", "Project Group"}
+- vCourseType1: courseType ∈ *validCourseTypes*
+- iCourseType1: courseType ∉ *validCourseTypes*
+
+### startDate (String)
+*validDates* := {"1000-01-01", ..., "9999-12-31" | x is an existing date }
+- vStartDate1: startDate ∈ *validDates*
+- iStartDate1: startDate ∉ *validDates* BUT is of format YYYY-MM-DD
+- iStartDate2: startDate is not of the format YYYY-MM-DD
+
+### endDate (String)
+- vEndDate1: endDate ∈ *validDates*
+- iEndDate1: endDate ∉ *validDates* BUT is of format YYYY-MM-DD
+- iEndDate2: endDate is not of the format YYYY-MM-DD
+
+### ects (Int)
+- vEcts1: ects ∈ [1, 999]
+- iEcts1: ects ∉ [1, 999]
```

Comment:
we should add courses with 0 ects for certain students (C++ course for CS students)

Reply:
While I agree with that, this PR is not the correct place for this request, as this PR handles only moving the data to the api